### DISCUSSION TITLE
Help with Django integration

### DISCUSSION THREAD

**Discussion by ognjenk** [ag2ai/faststream#1136] [Q&A] [answered]
Hi,
I have followed the Django integration guidelines outlined in the docs: https://faststream.airt.ai/latest/getting-started/integrations/django/?h=django, just replaced Kafka with RabbitMq broker as that is what I am using.

This is the code:

```python
import os
from contextlib import asynccontextmanager

from typing import AsyncIterator, TypedDict

from django.core.asgi import get_asgi_application
from starlette.applications import Starlette
from starlette.routing import Mount

from faststream.rabbit import RabbitBroker

os.environ.setdefault("DJANGO_SETTINGS_MODULE", "app.settings")

broker = RabbitBroker()

@asynccontextmanager
async def broker_lifespan(app):
    await broker.start()
    try:
        yield
    finally:
        await broker.close()


app = Starlette(
    routes=(
        Mount("/", get_asgi_application()),
    ),
    lifespan=broker_lifespan,
)

```

This works ok and the process establishes the connection to RabbitMq.
The question now is, how do I reference this broker from Django views? I would like to publish messages when a view is accessed.
I am running under gunicorn with uvicorn workers, getting no errors on startup and all views are working fine.

Thanks in advance!

**Comment by Lancetnik** [comment, marked as answer]
@ognjenk
I happy, that **FastStream** is helpful for you!
I think, the answer on your question can be found in #854 discussion
There you can find a little code snippet to make a broker syncify wrapper. This way you can easily publish messages from django sync code.

**Comment by ognjenk** [comment]
Ah perfect, somehow missed that answer!
Thank you very much and good luck with this library, it is proving to be very useful!
